plugins/lute-khmer/lute_khmer_parser/parser.py

```diff
@@ -0,0 +1,86 @@
+"""
+Parsing using khmer-nltk
+
+Includes classes:
+
+- KhmerParser
+
+"""
+
+import re
+
+from typing import List
+
+import khmernltk
+
+from lute.parse.base import ParsedToken, AbstractParser
+
+
+class KhmerParser(AbstractParser):
+    """
+    A parser for KHMER
+    """
+
+    @classmethod
+    def name(cls):
+        return "Lute Khmer"
+
+    @classmethod
+    def uses_data_directory(cls):
+        "Uses the data_directory (defined in the AbstractParser)."
+        return False  # or True
+
+    # @classmethod
+    # def init_data_directory(cls):
+    #     "Set up necessary files."
+    #     pass
+
+    def get_parsed_tokens(self, text: str, language) -> List[ParsedToken]:
+        """
+        Returns ParsedToken array for given language.
+        """
+
+        # Ensure standard carriage returns so that paragraph
+        # markers are used correctly.  Lute uses paragraph markers
+        # for rendering.
+        text = text.replace("\r\n", "\n")
+        text = text.replace("\n", "\\")
+
+        words = khmernltk.word_tokenize(text)  # ... get words using parser.
+        tokens = []
+        pattern = f"[{language.word_characters}]"
+        whitespace_regex = r"[ \t]+"
+        for word in words:
+            is_end_of_sentence = word in language.regexp_split_sentences
+            is_whitespace = re.match(whitespace_regex, word) is not None
+            if is_whitespace:
+                continue
+
+            is_word_char = (not is_end_of_sentence) and (
+                re.match(pattern, word) is not None
+            )
+
+            if word == "\\":
+                word = "¶"
+            if word == "¶":
+                is_word_char = False
+                is_end_of_sentence = True
+
+            if word.startswith("\\"):
+                num_leading_slashes = len(word) - len(word.lstrip("\\"))
+                for _ in range(num_leading_slashes):
+                    tokens.append(ParsedToken("¶", False, True))
+
+                word = word.lstrip("\\")
+                is_word_char = True
+                is_end_of_sentence = False
```

Comment:
Note about this logic here, the `khmernltk` parser completely gets rid of `\n` characters. What I decided to do instead is replace all the `\n` with `\\`, resulting in some words coming out as `\\ខ្ញុំ`. This logic replace any preceding slashes with the special character and then parses the original token like normal

Reply:
Good approach if it's working for you :-)